### PR TITLE
Add new tool: Sinto

### PR DESCRIPTION
**Pull request by pavanvidem**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by pavanvidem** [comment]
These tools can be used for scATAC-seq data preprocessing.

File: tools/sinto/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+categories: [Sequence Analysis]
+description: Sinto single-cell analysis tools
+homepage_url: https://github.com/timoast/sinto
+long_description: Sinto is a toolkit for processing aligned single-cell data. Used to add cell barcodes
```
Comment (bgruening) [inline tools/sinto/.shed.yml:4]: That should be a suite since you have multiple tools in here.

File: tools/sinto/sinto_barcode.xml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    <description>add cell barcodes to FASTQ read names</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">sinto</requirement>
+    </requirements>
+    <version_command>sinto --version</version_command>
+    <command><![CDATA[
+    cp $barcode_fastq barcodes.fastq.gz &&
```
Comment (bgruening) [inline tools/sinto/sinto_barcode.xml:11]: why not symlink?
Please use single quotes.

File: tools/sinto/sinto_barcode.xml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            </param>
+            <when value="paired">
+                <param name="read1_fastq" type="data" format="fastqsanger.gz" label="Forward reads FASTQ file" />
+                <param name="read2_fastq" type="data" format="fastqsanger.gz" label="Reverse reads FASTQ file" />
+            </when>
+            <when value="single">
+                <param name="read1_fastq" type="data" format="fastqsanger.gz" label="Select FASTQ file" />
+            </when>
+        </conditional>
+        <param type="integer" name="bases" value="16" label="Number of bases to extract from barcode-containing FASTQ" />
```
Comment (bgruening) [inline tools/sinto/sinto_barcode.xml:39]: min/max?

File: tools/sinto/sinto_fragments.xml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">sinto</requirement>
+    </requirements>
+    <version_command>sinto --version</version_command>
+    <command><![CDATA[
+    sinto fragments 
+    --bam $bam
```
Comment (bgruening) [inline tools/sinto/sinto_fragments.xml:12]: single-quotes

File: tools/sinto/sinto_fragments.xml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    --barcodetag '$barcodetag' 
+    --barcode_regex '${ str( $barcode_regex ) }'
+    --max_distance $max_distance
+    --min_distance $min_distance
+    --shift_plus $shift_plus
+    --shift_minus $shift_minus
+    $collapse_within
+    --fragments fragments.unsorted.tabular
+    --nproc "\${GALAXY_SLOTS:-1}" &&
+    sort -k 1,1 -k2,2n fragments.unsorted.tabular > fragments.tabular
```
Comment (bgruening) [inline tools/sinto/sinto_fragments.xml:23]: Do we need to require a gnu-coreutils requirement here?

File: tools/sinto/sinto_fragments.xml [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        <param argument="--barcodetag" type="text" value="CB" label="Read tag storing cell barcode information" />
+        <param argument="--barcode_regex" type="text" value="[^:]*" label="Read tag storing cell barcode information" 
+            help="[^:]* matches all characters up to the first colon" >
+            <sanitizer>
+                <valid initial="string.printable">
+                    <remove value="&apos;" />
+                </valid>
+            </sanitizer>
+        </param>
+        <param argument="--max_distance" type="integer" value="5000" label="Maximum distance between integration sites for the fragment to be retained"
```
Comment (bgruening) [inline tools/sinto/sinto_fragments.xml:37]: min/max?

File: tools/sinto/sinto_fragments.xml [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+        <param argument="--min_distance" type="integer" value="10" label="Minimum distance between integration sites for the fragment to be retained" 
+            help="Allows filtering of implausible fragments that likely result from incorrect mapping positions. Default is 5000 bp."/>
+        <param argument="--shift_plus" type="integer" value="4" label="Minimum MAPQ required to retain fragment" />
+        <param argument="--shift_minus" type="integer" value="-5" label="Minimum MAPQ required to retain fragment" />
+        <param argument="--collapse_within" type="boolean" truevalue="--collapse_within" falsevalue="" label="Take cell barcode into account 
+            when collapsing duplicate fragments" help="Setting this flag means that fragments with the same coordinates can be identified 
+            provided they originate from a different cell barcode." />
+    </inputs>
+    <outputs>
+        <data name='fragments' format='tabular' label="${tool.name} on ${on_string}: fragments" from_work_dir="fragments.tabular" />
```
Comment (bgruening) [inline tools/sinto/sinto_fragments.xml:48]: adding column names as metadata might be useful

**Review by bgruening — COMMENTED** [review]
bai files should not be included

File: tools/sinto/sinto_barcode.xml (hunk 2 above)
Comment (pavanvidem) [reply tools/sinto/sinto_barcode.xml]: I don't know. For some reason, I thought we are not using symlinks anymore. Now replaced all `cp` with `ln -s`.

File: tools/sinto/sinto_barcode.xml (hunk 3 above)
Comment (pavanvidem) [reply tools/sinto/sinto_barcode.xml]: I set the min to 0. max depends on the read length.

File: tools/sinto/sinto_fragments.xml (hunk 5 above)
Comment (pavanvidem) [reply tools/sinto/sinto_fragments.xml]: Removed the sort command. I don't think we need to sort. We can use Galaxy sorting tools.

File: tools/sinto/sinto_fragments.xml (hunk 6 above)
Comment (pavanvidem) [reply tools/sinto/sinto_fragments.xml]: min has been set. There is no theoretical max limit.

File: tools/sinto/sinto_fragments.xml (hunk 7 above)
Comment (pavanvidem) [reply tools/sinto/sinto_fragments.xml]: It is a bed file. I changed the format to bed.

**Review by bgruening — COMMENTED** [review]
Looks good to me, but tests are failing.

Please note that tests are running here with `planemo test --biocontainers`. That might help you to track down the error.

**Comment by pavanvidem** [comment]
It was complaining about BAM index. I symlinked the BAM and created an index. Do we need samtools as a requirement now?

**Comment by pavanvidem** [comment]
Success!

**Review by bgruening — APPROVED** [review]

**Pull request merged**